spec/features/remember_device/user_opted_preference.rb

```diff
@@ -0,0 +1,170 @@
+require 'rails_helper'
+
+describe 'Unchecking remember device' do
+  describe '2fa setup' do
+    context 'when the 2fa is totp' do
+      before do
+        user = sign_in_user
+
+        select_2fa_option('auth_app')
+
+        secret = find('#qr-code').text
+        fill_in 'code', with: generate_totp_code(secret)
+        uncheck 'remember_device'
+
+        click_button 'Submit'
+
+        click_continue
+        set_up_2fa_with_backup_code
+
+        first(:link, t('links.sign_out')).click
+        sign_in_user(user)
+      end
+
+      it 'requires the user to 2fa again upon sign in' do
+        expect(current_url).to include('/login/two_factor/authenticator')
+      end
+
+      it 'has an unchecked remember device checkbox upon next sign in' do
+        expect(page).to have_unchecked_field('remember_device')
+      end
+    end
+
+    context 'when the 2fa is webauthn' do
+      include WebAuthnHelper
+
+      before do
+        user = sign_in_user
+        mock_webauthn_setup_challenge
+        select_2fa_option('webauthn')
+
+        allow(WebauthnSetupForm).to receive(:domain_name).and_return('localhost:3000')
+
+        uncheck(:remember_device)
+        fill_in_nickname_and_click_continue
+
+        mock_press_button_on_hardware_key_on_setup
+        click_continue
+
+        set_up_2fa_with_backup_code
+
+        first(:link, t('links.sign_out')).click
+        sign_in_user(user)
+      end
+
+      it 'requires the user to 2fa again upon sign in' do
+        expect(current_url).to include('/login/two_factor/webauthn')
```

Comment:
If you wanted:

```ruby
        expect(current_url).to include('/login/two_factor/webauthn'), 'requires the user to 2fa again upon sign in'
```
so that the message stays clear if we combined the it-blocks